### PR TITLE
[build-command] Use uv to install Python deps for CI

### PR DESCRIPTION
**Pull request by echoix**
I just wanted to try it out with an upcoming PR, and it’s quite impressive.
With cached dependencies, the old way method took 25-40s to install the deps. (https://github.com/echoix/megalinter/actions/runs/9121261204/job/25080127542#step:8:1 and https://github.com/echoix/megalinter/actions/runs/9120700814/job/25078549631#step:8:1)

Here, WITHOUT cached dependencies, it took less than 5 secs (https://github.com/echoix/megalinter/actions/runs/9121318153/job/25080283053#step:9:1)

I used the pip to install uv, but it could’ve been with the shell script one liner.

I had to use the `--system` flag to let it use the global install in CI, and to not require a venv.

Otherwise, it’s quite easy.

## Proposed Changes

1. …
2. …
3. …

### Author/Contributor
- [ ] Add entry to the [CHANGELOG](https://github.com/oxsecurity/megalinter/blob/main/CHANGELOG.md) listing the change and linking to the corresponding issue (if appropriate)
- [ ] If documentation is needed for this change, has that been included in this pull request

### Reviewing Maintainer
- [ ] Label as `breaking` if this is a large fundamental change
- [ ] Label as either `automation`, `bug`, `documentation`, `enhancement`, `infrastructure`, or `performance`

### CODE REVIEW

**Comment by nvuillam** [comment]
@echoix that's amazing ^^

What about ascending dependency ? For example, what happens to repos using pip install in PRE_COMMANDS ?

If this is a breaking change, we can't merge this PR before the next major release

**Comment by echoix** [comment]
It is only inside the build command when used by us in a comment to rebuild the generated files on a PR.


- But to answer your question, the maintainer of the UV repo answered the other day that a venv populated by uv is undistinguishable from the venv's point of view, and can be further modified by it. So as a docker build step tool, it is ideal. But in this PR, it isn't in any docker script.

**Comment by echoix** [comment]
Since a required check was skipped, it will need to be force merged @nvuillam

**Review by nvuillam — APPROVED** [review]

**Comment by nvuillam** [comment]
@echoix great ')
Merged ^^

**Pull request merged**
